worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/implementation/ParallelQueueExtent.java

```diff
@@ -40,19 +45,30 @@
 
 public class ParallelQueueExtent extends PassthroughExtent implements IQueueWrapper {
 
+    private static final Logger LOGGER = LogManagerCompat.getLogger();
+
     private final World world;
     private final QueueHandler handler;
     private final BatchProcessorHolder processor;
     private final BatchProcessorHolder postProcessor;
+    private final boolean[] faweExceptionReasonsUsed = new boolean[FaweException.Type.values().length];
     private int changes;
     private final boolean fastmode;
+    private int lastException = Integer.MIN_VALUE;
+    private int exceptionCount = 0;
 
     public ParallelQueueExtent(QueueHandler handler, World world, boolean fastmode) {
         super(handler.getQueue(world, new BatchProcessorHolder(), new BatchProcessorHolder()));
         this.world = world;
         this.handler = handler;
         this.processor = (BatchProcessorHolder) getExtent().getProcessor();
+        if (this.processor.getProcessor() instanceof MultiBatchProcessor) {
+            ((MultiBatchProcessor) this.processor.getProcessor()).setFaweExceptionArray(faweExceptionReasonsUsed);
```

Comment:
Is it thread safe to share the array here?

Reply:
See above